pallets/transaction-multi-payment/src/lib.rs

```diff
@@ -202,15 +260,33 @@ impl<T: Config> Module<T> {
 
 		// If not native currency, let's buy CORE asset first and then pay with that.
 		if fee_currency != CORE_ASSET_ID {
-			T::AMMPool::buy(&who, CORE_ASSET_ID, fee_currency, fee, 2u128 * fee, false)?;
+			T::AMMPool::buy(&who, AssetPair{asset_out: CORE_ASSET_ID, asset_in: fee_currency}, fee, 2u128 * fee, false)?;
```

Comment:
`2u128 * fee` magic swap limit, this should be mby explained in some function and preferably it would be lower than twice the expected amount... mby this should be configurable by user in future

Reply:
I think configurable option and can be set together with set_currency, so user  can set the limit  alongside chosen currency, is quite a good approach.

@jak-pan what do you think ?

Reply:
What I am thinking is that we might want to simplify this sooner than expected since this brings so many questions and probably performance implications. What we wanted to do with tx fees is to burn them instead of accumulating. Similar to EIP1559, but this brings some tokenomics questions which we didn't clear yet. 

Right now, we're doing User -> poolSwap (+token,-HDX) -> treasury. We could do just the burn and send on some very basic math really. 

Soo... The solution for me is to make it as simple as possible and create new issue that targets this.

Reply:
Question is can this be done in normal pools as well?